### DISCUSSION TITLE
ScriptReference/ObjectChangeKind.ChangeGameObjectStructure

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#672] [Notes]
# ScriptReference/ObjectChangeKind.ChangeGameObjectStructure

https://docs.unity3d.com/ScriptReference/ObjectChangeKind.ChangeGameObjectStructure.html

**Comment by andreasIsCoding** [comment]
This callback will only return the Game Object.  Use `ObjectFactory.componentWasAdded` to get the actual component that was added.  Since componentWasAdded fires first, you can can save a reference to the Game Object the component was attached to, and test that to discover if the game object had a component removed (and not added).
https://docs.unity3d.com/ScriptReference/ObjectFactory-componentWasAdded.html
